internal/controllers/topology/cluster/conditions.go

```diff
@@ -119,6 +120,10 @@ func (r *Reconciler) reconcileTopologyReconciledCondition(s *scope.Scope, cluste
 			reason = clusterv1.TopologyReconciledMachineDeploymentsUpgradeDeferredReason
 		}
 
+		isAnyMachineDeploymentsUpgrading, err := s.Current.MachineDeployments.IsAnyUpgrading(ctx, r.Client)
```

Comment:
Q: Can we infer IsAnyUpgrading from the s.UpgradeTracker instead of reading again machines? (We store what is upgrading when calling MarkUpgrading)

Reply:
+1 would be good to base this on the same data. If we run this again the state of the MDs in the cluster might have changed already